pyvo/dal/exceptions.py

```diff
@@ -176,11 +176,18 @@ def from_except(cls, exc, url=None):
         for the given exception that represents the underlying cause.
         """
         if isinstance(exc, requests.exceptions.RequestException):
-            message = str(exc)
             try:
-                code = exc.response.status_code
+                response = exc.response
             except AttributeError:
-                code = 0
+                response = None
+            code = 0
+            if response is not None:
+                code = response.status_code
+            message = str(exc)
+            content_type = response.headers.get('content-type', None)
+            if content_type and 'text/plain' in content_type:
+                message = '{} for {}'.format(response.text, url)
```

Comment:
so in this case you drop the message from the exception?

Reply:
Yes, it makes a subtle assumption that this is a `requests.HttpError` exception and as such the original exception message (if any) is very generic (`No Found`, etc) but has no much value. The other assumption here is that the payload has more details.

Reply:
Thanks, that sounds great!